lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -0,0 +1,651 @@
+""" API v0 views. """
+
+import datetime
+import logging
+import pytz
+
+from django.contrib.auth.models import User
+from django.db import transaction
+from django.http import Http404
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from ccx_keys.locator import CCXLocator
+from courseware import courses
+from instructor.enrollment import (
+    enroll_email,
+    get_email_params,
+)
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.lib.api.permissions import IsCourseInstructor
+from student.models import CourseEnrollment
+from student.roles import CourseCcxCoachRole
+
+
+from lms.djangoapps.ccx.models import CcxFieldOverride, CustomCourseForEdX
+from lms.djangoapps.ccx.overrides import (
+    override_field_for_ccx,
+)
+from lms.djangoapps.ccx.utils import (
+    assign_coach_role_to_ccx,
+    is_email,
+)
+from .paginators import CCXAPIPagination
+from .serializers import CCXCourseSerializer
+
+log = logging.getLogger(__name__)
+TODAY = datetime.datetime.today  # for patching in tests
+
+
+def get_valid_course(course_id, is_ccx=False, advanced_course_check=False):
+    """
+    Helper function used to validate and get a course from a course_id string.
+    It works with both master and ccx course id.
+
+    Args:
+        course_id (str): A string representation of a Master or CCX Course ID.
+        is_ccx (bool): Flag to perform the right validation
+        advanced_course_check (bool): Flag to perform extra validations for the master course
+
+    Returns:
+        tuple: a tuple of course_object, course_key, error_code, http_status_code
+    """
+    if course_id is None:
+        # the ccx detail view cannot call this function with a "None" value
+        if not is_ccx:
+            log.info('Master course ID not provided')
+            error_code = 'master_course_id_not_provided'
+
+        return None, None, error_code, status.HTTP_400_BAD_REQUEST
```

Comment:
In theory this function is never called with `is_ccx=True` and no `course_is` because the view for the CCX details enforces the presence of the course id (the course key string), but I see your point and I will add a default `error_code` string to cover the edge case.